docs/guide/pug-specific-options.md

```diff
@@ -0,0 +1,453 @@
+
+Closing bracket ends with a new line.
+
+```pug
+input(
+  type="text",
+  value="my_value",
+  name="my_name",
+  alt="my_alt",
+  autocomplete="on"
+)
+```
+
+#### `'last-line'`
+
+Closing bracket remains with last attribute's line.
+
+```pug
+input(
+  type="text",
+  value="my_value",
+  name="my_name",
+  alt="my_alt",
+  autocomplete="on")
+```
+
+---
+
+## Comment Preserve Spaces
+
+`pugCommentPreserveSpaces`
+
+### Description
+
+Change behavior of spaces within comments.
+
+### Options
+
+**Type:** `choice`  
+**Default:** `'keep-all'`
+
+#### `'keep-all'`
+
+Keep all spaces within comments.
+
+```pug
+// ___this _is __a __comment_
+```
+
+#### `'keep-leading'`
+
+Keep leading spaces within comments.
+
+```pug
+// ___this is a comment
+```
+
+#### `'trim-all'`
+
+Trim all spaces within comments.
+
+```pug
+// this is a comment
+```
+
+---
+
+## Sort Attributes
+
+`pugSortAttributes`
+
+### Description
+
+Sort attributes that are not on _beginning_ and _end_ patterns.
+
+### Options
+
+**Type:** `choice`  
+**Default:** `'as-is'`
+
+#### `'as-is'`
+
+Keep the attributes untouched.
+
+```pug
+Foo(a c d b)
+```
+
+#### `'asc'`
+
+Sort attributes ascending.
+
+```pug
+Foo(a b c d)
+```
+
+#### `'desc'`
+
+Sort attributes descending.
+
+```pug
+Foo(d c b a)
+```
+
+---
+
+`pugSortAttributesBeginning`
+
+### Description
+
+Sort attributes by regex patterns to the beginning.
+
+### Options
+
+**Type:** `array`  
+**Default:** `[]`
+
+---
+
+`pugSortAttributesEnd`
+
+### Description
+
+Sort attributes by regex patterns to the end.
+
+### Options
+
+**Type:** `array`  
+**Default:** `[]`
+
+---
+
+[Example](https://github.com/prettier/plugin-pug/issues/22#issuecomment-699509995)
+
+---
+
+## Wrap Attributes
+
+`pugWrapAttributesPattern`
+
+### Description
+
+Define a regex pattern to match attributes against that should always trigger wrapping.
+
+### Options
+
+**Type:** `array`  
+**Default:** `[]`
+
+---
+
+`pugWrapAttributesThreshold`
+
+### Description
+
+Define the maximum amount of attributes that an element can appear with on one line before it gets wrapped.
+
+### Options
+
+**Type:** `choice`  
+**Default:** `-1`
+
+#### `-1`
+
+Only wrap attributes as needed.
+
+```pug
+input(type="text")
+input(type="text", value="my_value", name="my_name")
+```
+
+#### `0`
+
+Always wrap attributes.
+
+```pug
+input(
+  type="text"
+)
+input(
+  type="text",
+  value="my_value",
+  name="my_name"
+)
+```
+
+#### `1`
+
+Allow one unwrapped attribute, wrap two and more.
+
+```pug
+input(type="text")
+input(
+  type="text",
+  value="my_value",
+  name="my_name"
+)
+```
+
+#### `2 .. Infinity`
+
+Same as above, just with different thresholds.
+
+---
+
+## Framework specific
+
+`pugFramework`
+
+### Description
+
+Specify the used framework within the project.  
+If not set, or set to 'none', the plugin tries to find the correct framework by reading `process.env.npm_package_*`.
+
+### Options
+
+**Type:** `choice`  
+**Default:** `'none'`
```

Comment:
Probably off topic, but maybe `auto` or `detect` would be a better value for "auto detect".

Reply:
I also thought about that, but it would be a code change instead of a docs change